Refresh.GameServer/Resources/ResourceHelper.cs

```diff
@@ -9,19 +10,57 @@ namespace Refresh.GameServer.Resources;
 
 public static class ResourceHelper
 {
-    public static Stream StreamFromResource(string name)
+    public static Stream StreamFromResource(string name, Assembly? assembly = null)
     {
-        Assembly assembly = Assembly.GetExecutingAssembly();
+        assembly ??= Assembly.GetExecutingAssembly();
         return assembly.GetManifestResourceStream(name)!;
     }
+
+    public static byte[] ReadResource(string name, Assembly? assembly = null)
+    {
+        using MemoryStream stream = new();
+        StreamFromResource(name, assembly).CopyTo(stream);
+        return stream.ToArray();
+    }
     
     private static readonly ThreadLocal<Iron> Iron = new(() => new Iron(ArrayPool<byte>.Shared));
 
-    private static void Encrypt(ReadOnlySpan<byte> data, ReadOnlySpan<byte> key)
+    public static byte[] PspEncrypt(Span<byte> data, ReadOnlySpan<byte> key)
     {
-        throw new NotImplementedException(); //TODO
+        byte[] initialCounter = new byte[16];
+        key[..8].CopyTo(initialCounter);
+        
+        //If the file is big enough, try to compress it
+        bool compress = data.Length > 1024;
+
+        //If we want to compress the data, do so, if not, dont
+        ReadOnlySpan<byte> resultData = compress
+            ? Iron.Value!.Compress(Codec.LZO, data, null, CompressionLevel.SmallestSize).AsSpan().ToArray()
```

Comment:
So... we have a byte array from `Compress`. Then, we cast it to a span. Then we convert the span to a byte array. Finally, we store it as a ReadOnlySpan<byte>.

What?

Reply:
Compress does not return a byte array, its a special data type, calling `AsSpan()` gets the span form of the underlying data, then calling `ToArray` makes a copy, since we only need to read from that copy, we then store that into a `ReadOnlySpan<byte>`
In this case though we can skip the copy step, since we dont actually return `resultData` instead copying it into `final`, i'll remove that extra `ToArray`

Reply:
Should also make consts for the byte values down below (e.g. 0x19, 0x10, etc.) below just to give them some names.